crates/gen-wasmtime-py/src/lib.rs

```diff
@@ -558,12 +193,52 @@ impl WasmtimePy {
             self.src.push_str(&param.to_snake_case());
             params.push(param.to_snake_case());
             self.src.push_str(": ");
-            self.print_ty(iface, ty);
+            self.print_ty(iface, ty, true);
         }
         self.src.push_str(") -> ");
-        self.print_ty(iface, &func.result);
+        self.print_ty(iface, &func.result, true);
         params
     }
+
+    fn type_union_wrapped(&mut self, iface: &Interface, name: &str, union: &Union) {
+        self.pyimport("dataclasses", "dataclass");
+        let mut cases = Vec::new();
+        let name = name.to_camel_case();
+        for (i, case) in union.cases.iter().enumerate() {
+            self.docs(&case.docs);
+            self.src.push_str("@dataclass\n");
+            let name = format!("{name}{i}");
+            self.src.push_str(&format!("class {name}:\n"));
+            self.indent();
+            self.src.push_str("value: ");
+            self.print_ty(iface, &case.ty, true);
+            self.src.push_str("\n");
+            self.deindent();
+            self.src.push_str("\n");
+            cases.push(name);
+        }
+
+        self.pyimport("typing", "Union");
+        self.src
+            .push_str(&format!("{name} = Union[{}]\n", cases.join(", ")));
+        self.src.push_str("\n");
+    }
+
+    fn type_union_raw(&mut self, iface: &Interface, name: &str, union: &Union) {
+        self.pyimport("typing", "Union");
+        self.src.push_str(&name.to_camel_case());
+        self.src.push_str(" = Union[");
+        let mut first = true;
+        for case in union.cases.iter() {
+            self.docs(&case.docs);
```

Comment:
Does this work having docs inline here?

Reply:
It looks like it was passing tests because despite being very janky, it did in fact generate valid code.
```py
# Overview
Num = Union[# Case 0
float# Case 1
,int]
```
I've updating the code to generate this instead, which is a bit more sane.
```py
# Overview
# Case 0
# Case 1
Num = Union[float,int]
```